### ISSUE TITLE
Cannot dynamically change locale - I18n.of(context).locale = Locale("pt_BR") not working

### ISSUE THREAD

**Issue by SebastienMelki** [marcglasberg/i18n_extension#5]
Hello. Thanks for the great plugin!

I'm having issues reloading the i18n locale programatically. If I change the language and hot reload the app everything works fine, but I basically want all the translations to change to the chosen language on the tap of a button. 
I used I18n.of(context).locale = Locale($MY_NEW_LOCALE); like the docs suggested but I get the following error, and my strings don't change language... any idea how I can proceed?
I'm changing from english to arabic, so when I tap my change language button the entire app is becoming RTL (base locale is changing) but the translations don't change until I reload the app. 

Thanks for the help 

`I/flutter (17829): ══╡ EXCEPTION CAUGHT BY GESTURE ╞═══════════════════════════════════════════════════════════════════
I/flutter (17829): The following NoSuchMethodError was thrown while handling a gesture:
I/flutter (17829): The getter 'data' was called on null.
I/flutter (17829): Receiver: null
I/flutter (17829): Tried calling: data
I/flutter (17829): 
I/flutter (17829): When the exception was thrown, this was the stack:
I/flutter (17829): #0      Object.noSuchMethod (dart:core-patch/object_patch.dart:53:5)
I/flutter (17829): #1      I18n.of (package:i18n_extension/i18n_widget.dart:74:73)
I/flutter (17829): #2      _SettingsPageState.languageButton.<anonymous closure> (package:anghamicreators/widgets/settings_page.dart:92:18)
I/flutter (17829): #3      _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:706:14)
I/flutter (17829): #4      _InkResponseState.build.<anonymous closure> (package:flutter/src/material/ink_well.dart:789:36)
I/flutter (17829): #5      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:182:24)
I/flutter (17829): #6      TapGestureRecognizer.handleTapUp (package:flutter/src/gestures/tap.dart:486:11)
I/flutter (17829): #7      BaseTapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:264:5)
I/flutter (17829): #8      BaseTapGestureRecognizer.acceptGesture (package:flutter/src/gestures/tap.dart:236:7)
I/flutter (17829): #9      GestureArenaManager.sweep (package:flutter/src/gestures/arena.dart:156:27)
I/flutter (17829): #10     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:222:20)
I/flutter (17829): #11     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:198:22)
I/flutter (17829): #12     GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:156:7)
I/flutter (17829): #13     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:102:7)
I/flutter (17829): #14     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:86:7)
I/flutter (17829): #18     _invoke1 (dart:ui/hooks.dart:273:10)
I/flutter (17829): #19     _dispatchPointerDataPacket (dart:ui/hooks.dart:182:5)
I/flutter (17829): (elided 3 frames from package dart:async)
I/flutter (17829): 
I/flutter (17829): Handler: "onTap"
I/flutter (17829): Recognizer:
I/flutter (17829):   TapGestureRecognizer#bef9f
I/flutter (17829): ════════════════════════════════════════════════════════════════════════════════════════════════════`

**Comment by marcglasberg** [comment]
@SebastienMelki You need to provide me with a minimal runnable example. A single file with a main method for me to run and see the problem. Can you do that for me, please?

**Comment by SebastienMelki** [comment]
Hey @marcglasberg thank you so much for replying. I built a small example to showcase what was going wrong but everything worked fine. So I went back to my initial project and debugged widget by widget, and turns out there's a bad interaction with the following plugin https://pub.dev/packages/splashscreen that I was using to display a splash screen at app launch.

I simply stopped using the plugin and everything worked fine! I'll have to implement splash screen natively :D 

Thanks again!
Cheers
